File: app/src/main/java/com/fpoly/project1/firebase/model/Customer.java
package com.fpoly.project1.firebase.model;

import com.google.firebase.database.IgnoreExtraProperties;

import java.util.List;

@IgnoreExtraProperties
public class Customer {
    public String __id;
    public String gid;
    public String fid;
    public String avatarUrl;
    public String fullName;
    public String birthDate;
    public String emailAddress;
    public String phoneNumber;
    public String postalAddress;
    public List<String> favoriteIds;

    public Customer() {
    }

    /**
     * @param __id          Firebase unique ID
     * @param gid           Google account unique ID
     * @param fid           Facebook account unique ID
     * @param avatarUrl     Account public avatar url, use Glide to load the image
     * @param fullName      Account display name
     * @param birthDate     Account birth date, optionally set by user via account settings screen
     * @param emailAddress  Account email address
     * @param phoneNumber   Account phone number
     * @param postalAddress Account postal address, set by user via account creation screen
     * @param favoriteIds   Account list of favorite products (by product ID)
     */
    public Customer(String __id, String gid, String fid, String avatarUrl, String fullName, String birthDate, String emailAddress, String phoneNumber, String postalAddress, List<String> favoriteIds) {
        this.__id = __id;
        this.gid = gid;
        this.fid = fid;
        this.avatarUrl = avatarUrl;
        this.fullName = fullName;
        this.birthDate = birthDate;
        this.emailAddress = emailAddress;
        this.phoneNumber = phoneNumber;
        this.postalAddress = postalAddress;
        this.favoriteIds = favoriteIds;
    }
}

File: app/src/main/java/com/fpoly/project1/firebase/model/Customer.kt
package com.fpoly.project1.firebase.model

import com.fpoly.project1.firebase.Firebase
import com.fpoly.project1.firebase.controller.ControllerBase

data class Customer(
    var id: String? = null,
    var gid: String? = null,
    var fid: String? = null,
    var avatarUrl: String? = null,
    var fullName: String? = null,
    var birthDate: String? = null,
    var emailAddress: String? = null,
    var phoneNumber: String? = null,
    var postalAddress: String? = null,
    var favoriteIds: List<String>? = null
) {
    fun getAvatarUrl(successListener: ControllerBase.SuccessListener) {
        Firebase.storage.child("/avatars/${this.id}.jpg").downloadUrl
            .addOnCompleteListener {
                if (it.isSuccessful)
                    successListener.run(it.result.toString())
                else
                    successListener.run(this.avatarUrl)
            }
    }
}
